### PR TITLE
Added new internal api for fetching list of user profiles.

### PR DESCRIPTION
**Pull request by lovoll**
## Description
New internal API for fetching user profiles as a list.

## Related Issue(s)
- #129 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [x] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Comment by SandGrainOne** [comment]
One failing test when it's running on the build agent:
  Failed
```
Altinn.Profile.Tests.IntegrationTests.UserProfileInternalTests.GetUserListByUuid_SblBridgeFindsProfile_ResponseOk_ReturnsUserProfileList [43 ms]
Error Message:
Assert.Equal() Failure: Strings differ
                   ↓ (pos 8)
Expected: "ELENA FJ�R"
Actual:   "ELENA FJÆR"
                   ↑ (pos 8)
```

File: test/Altinn.Profile.Tests/Testdata/UserProfile/4c3b4909-eb17-45d5-bde1-256e065e196a.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "PhoneNumber": null,
+  "Email": null,
+  "PartyId": 50002114,
+  "Party": {
+    "PartyTypeName": 1,
+    "SSN": "01025161013",
+    "OrgNumber": "",
+    "Person": {
+      "SSN": "01025161013",
+      "Name": "ELENA FJÆR",
```
Comment (SandGrainOne) [inline test/Altinn.Profile.Tests/Testdata/UserProfile/4c3b4909-eb17-45d5-bde1-256e065e196a.json:16]: Test is failing because of this.

Comment (SandGrainOne) [reply test/Altinn.Profile.Tests/Testdata/UserProfile/4c3b4909-eb17-45d5-bde1-256e065e196a.json:16]: Good test :)

Comment (lovoll) [reply test/Altinn.Profile.Tests/Testdata/UserProfile/4c3b4909-eb17-45d5-bde1-256e065e196a.json]: Could not find out why the Æ was encoded wrong on build machine when it was stored directly in source code. It was working on Dev machine so ended up just ignoring to test this and removed all Æs from the test.

File: test/Altinn.Profile.Tests/UnitTests/UserProfileCachingDecoratorTest.cs [hunk 2]
```diff
@@ -167,6 +263,29 @@ public async Task GetUserUserUserUuid_NullFromDecoratedService_CacheNotPopulated
             Assert.False(memoryCache.TryGetValue($"User:UserUuid:{userUuid}", out UserProfile _));
         }
 
+        /// <summary>
+        /// Tests that if the result from decorated service is an empty list, nothing is stored in cache and the empty list returned to caller.
+        /// </summary>
+        [Fact]
+        public async Task GetUserListUserUserUuid_NullFromDecoratedService_CacheNotPopulated()
```
Comment (SandGrainOne) [inline test/Altinn.Profile.Tests/UnitTests/UserProfileCachingDecoratorTest.cs:270]:
```suggestion
        public async Task GetUserListUserUserUuid_EmptyListFromDecoratedService_CacheNotPopulated()
```

**Review by SandGrainOne — APPROVED** [review]

File: test/Altinn.Profile.Tests/UnitTests/UserProfileCachingDecoratorTest.cs (hunk 2 above)
Comment (lovoll) [reply test/Altinn.Profile.Tests/UnitTests/UserProfileCachingDecoratorTest.cs]: Changed

**Pull request merged**